Tests/WorkspaceTests/InitTests.swift

```diff
@@ -14,458 +14,422 @@ import Basics
-        try testWithTemporaryDirectory { tmpPath in
-            let fs = localFileSystem
-            let path = tmpPath.appending("MyBuildToolPlugin")
-            let name = path.basename
-            try fs.createDirectory(path)
 
-            // Create the package
-            try InitPackage(
-                name: name,
-                packageType: .buildToolPlugin,
-                destinationPath: path,
-                fileSystem: localFileSystem
-            ).writePackageStructure()
+    /// Tests creating a command plugin package.
+    @Test func initPackageCommandPlugin() async throws {
+        try await createAndVerifyPackage(
+            packageType: .commandPlugin,
+            name: "MyCommandPlugin",
+            supportedTestingLibraries: [],
+            customVerification: { path, name in
+                try verifyPluginPackage(at: path, name: name, isCommandPlugin: true)
+            }
+        )
+    }
 
-            // Verify basic file system content that we expect in the package
-            let manifest = path.appending("Package.swift")
-            XCTAssertFileExists(manifest)
-            let manifestContents: String = try localFileSystem.readFileContents(manifest)
-            XCTAssertMatch(manifestContents, .and(.contains(".plugin("), .contains("targets: [\"MyBuildToolPlugin\"]")))
-            XCTAssertMatch(manifestContents, .and(.contains(".plugin("), .contains("capability: .buildTool()")))
-
-            // Check basic content that we expect in the plugin source file
-            let source = path.appending("Plugins", "MyBuildToolPlugin.swift")
-            XCTAssertFileExists(source)
-            let sourceContents: String = try localFileSystem.readFileContents(source)
-            XCTAssertMatch(sourceContents, .contains("struct MyBuildToolPlugin: BuildToolPlugin"))
-            XCTAssertMatch(sourceContents, .contains("createBuildCommands(context: PluginContext"))
-            XCTAssertMatch(sourceContents, .contains("import XcodeProjectPlugin"))
-            XCTAssertMatch(sourceContents, .contains("extension MyBuildToolPlugin: XcodeBuildToolPlugin"))
-            XCTAssertMatch(sourceContents, .contains("createBuildCommands(context: XcodePluginContext"))
-        }
+    /// Tests creating a build tool plugin package.
+    @Test func initPackageBuildToolPlugin() async throws {
+        try await createAndVerifyPackage(
+            packageType: .buildToolPlugin,
+            name: "MyBuildToolPlugin",
+            supportedTestingLibraries: [],
+            customVerification: { path, name in
+                try verifyPluginPackage(at: path, name: name, isCommandPlugin: false)
+            }
+        )
     }
 
-    // MARK: Special case testing
+    // MARK: - Special Case Tests
 
-    func testInitPackageNonc99Directory() async throws {
+    /// Tests creating a package in a directory with a non-C99 compliant name.
+    @Test(arguments: [BuildSystemProvider.Kind.native, .swiftbuild])
```

Comment:
suggestion: consider passing the Build configuration as a test argument.  
```suggestion
    @Test(
        arguments: getBuildData(for: SupportedBuildSystemOnAllPlatforms),
    )
```